src/components/popup/details/details.jsx

```diff
@@ -28,24 +28,25 @@ export default class Details extends Component {
 		return Object.values(vendors);
 	};
 
-	filterVendors = ({ isCustom = null, purposeIds = [], featureIds = [] } = {}) => {
-		return this.getVendors().filter(vendor => {
-			if (isCustom !== null && (isCustom && vendor.globalId || !isCustom && !vendor.globalId)) {
-				return false;
+	filterVendors = ({ isCustom = null, purposeId, featureId, isSpecial} = {}) => {
```

Comment:
When filtering by purpose, we need to check two cases:
- if vendor has it in purposes array
- if vendor has it in legIntPurposes array

Now, if vendor has purposes: [1, 3] and legIntPurposes: [2] it won't be selected for purpose with id 2.

- [ ] use getVendorsWithLegIntPurpose function